New Guide: Self-hosted to RS-hosted data plane (#742)
rudderlabs/rudderstack-docs: docs/user-guides/how-to-guides/switch-to-rudderstack-hosted-data-plane.mdx

---
# slug: "/docs/user-guides/how-to-guides/switch-to-rudderstack-hosted-data-plane"
title: "How to Switch from Self-hosted to RudderStack-hosted Data Plane"
description: >-
  Step-by-step guide on switching from self-hosted data plane to RudderStack-hosted data plane.
---

The steps in this guide assume that you are using the RudderStack-hosted control plane and lists out the detailed steps to switch from self-hosted data plane to the RudderStack-hosted data plane.

## Self-hosted to RudderStack-hosted data plane 

Follow the steps below to switch from self-hosted to RudderStack-hosted data plane:

1. Update the `dataPlaneUrl` in all the SDKs to point to the RudderStack-hosted hosted data plane URL. Refer to the <Link to="/sources/sdks/">SDK documentation</Link> for details on how to obtain the data plane URL.

2. Verify that the self-hosted data plane is not getting any requests.

> **{{Steps, if any}}**

Reviewer: We can recommend them to check their load balancer metrics to verify if they are still receiving any requests on the self hosted on. Exact steps would vary on how they did the deployment

3. Verify that all the events received in the self-hosted data plane are delivered. 

> **{{`rudder-cli` commands for this}}**

Reviewer: we do not have a straight forward way to determine if all events are delivered by running a single command. We can recommend users to keep it running for a day or two and shut it down

4. Shut down the self-hosted data plane.

> **{{Steps, if any}}**

Reviewer: This would again depend on how they deployed. One example would be `helm delete <rudderstatck-release-name>` if they deployed using helm

## Best practices

* In case of mobile SDKs, your customers need to update their mobile applications. You can keep the self-hosted data plane online until a sufficient number of customers have updated their mobile application or there are no more events coming in to the self-hosted data plane.

  Alternatively, you can also add an alias DNS record for your self-hosted data plane URL to point to the RudderStack-hosted data plane URL.

> **{{Steps to add an alias DNS record, if any}}**

Author: @psrikanth88 Got it, I will add a cross-reference link here once we create/update the doc for the above-mentioned approaches.

---
# slug: "/docs/user-guides/how-to-guides/switch-to-rudderstack-hosted-data-plane"
title: "How to Switch from Self-hosted to RudderStack-hosted Data Plane"
description: >-
  Step-by-step guide on switching from self-hosted data plane to RudderStack-hosted data plane.
---

> The steps in this guide assume that you are using the RudderStack-hosted control plane and lists out the detailed steps to switch from self-hosted data plane to the RudderStack-hosted data plane.

Reviewer 2: "...the detailed steps to switch from self-hosted data plane to the RudderStack-hosted data plane." feels repetitive in the title, meta, intro, as well as the intro of the next section. 

Maybe we can start the guide with the following suggestion?

```
<div class="warningBlock">
The steps in this guide assume that you are using the RudderStack-hosted control plane.
</div>
```

## Self-hosted to RudderStack-hosted data plane 

Follow the steps below to switch from self-hosted to RudderStack-hosted data plane:

> 1. Update the `dataPlaneUrl` in all the SDKs to point to the RudderStack-hosted hosted data plane URL. Refer to the <Link to="/sources/sdks/">SDK documentation</Link> for details on how to obtain the data plane URL.

Reviewer 2: Can we add the reference to [this FAQ](https://www.rudderstack.com/docs/get-started/rudderstack-open-source/data-plane-setup/#what-is-a-data-plane-url-where-do-i-get-it) instead for info on obtaining the data plane URL?

Reviewer 2: Added suggestion below for more context

> 1. Update the `dataPlaneUrl` in all the SDKs to point to the RudderStack-hosted hosted data plane URL. Refer to the <Link to="/sources/sdks/">SDK documentation</Link> for details on how to obtain the data plane URL.

Reviewer 2: ```suggestion
1. Update the `dataPlaneUrl` in all your SDK configurations to point to the RudderStack-hosted hosted data plane URL. Refer to the specific <Link to="/sources/sdks/">SDK documentation</Link> or this <Link to="/get-started/rudderstack-open-source/data-plane-setup/#what-is-a-data-plane-url-where-do-i-get-it">FAQ</Link> for details on obtaining the data plane URL.
```

2. Verify that the self-hosted data plane is no longer getting any requests. One way is to check the load balancer metrics to verify if any requests are still received on your data plane.

3. Ensure that all the events received in the self-hosted data plane are delivered. 

<div class="infoBlock">
It is recommended to keep the self-hosted plane running for a day or two to ensure that all the events are delievered.
</div>

> 4. Shut down the self-hosted data plane dpending on the way of your deployment. For example, if you have deployed the data plane using helm, run the following command:

Reviewer 2: ```suggestion
4. Shut down the self-hosted data plane depending on the way of your deployment. For example, if you have deployed the data plane using helm, run the following command:
```

```BASH
helm delete <rudderstack-release-name>
```

<div class="infoBlock">

> Replace the <code class="inline-code">rudderstack-release-name</code> with the release name of the chart you want to delete.

Reviewer 2: ```suggestion
Replace the <code class="inline-code">rudderstack-release-name</code> with the release name of the Helm chart you want to delete.
```

</div>

## Best practices

> * In case of mobile SDKs, your customers need to update their mobile applications. You can keep the self-hosted data plane online until a sufficient number of customers have updated their mobile application or there are no more events coming in to the self-hosted data plane.

Reviewer 2: "..need to update their mobile applications."

Do we mean "your customers might also need to update the RudderStack implementation in their mobile apps."?

> * In case of mobile SDKs, your customers need to update their mobile applications. You can keep the self-hosted data plane online until a sufficient number of customers have updated their mobile application or there are no more events coming in to the self-hosted data plane.

Reviewer 2: ```suggestion
* In case of mobile SDKs, your customers need to update their mobile applications. You can keep the self-hosted data plane online until a satisfactory number of customers have updated their mobile application or there are no more events coming in to the self-hosted data plane.
```

Reviewer 2: not a very important suggestion but gives a sense that it's completely up to the reader to set this number.

Alternatively, you can run an API Gateway or Cloudflare workers to proxy the requests from your self-hosted data plane URL to the RudderStack-hosted data plane URL.

* In case of mobile SDKs, if you are using Firebase to serve the values like `writeKey` and `dataPlaneUrl`, there is no need to update the mobile application to point to the RudderStack-hosted data plane.

> * This process of switching the data plane can hamper the order of events as the events are sent both from the RudderStack-hosted and self-hosted data plane at the same time for a while. Hence, it is recommended to choose a low-traffic time slot to carry out this activity.

Reviewer 2: ```suggestion
* This process of switching the data plane can hamper the order of events as the events are sent both from the RudderStack-hosted and self-hosted data plane at the same time **for a while**. Hence, it is recommended to choose a low-traffic time slot to carry out this activity.
```

Reviewer 2: IMO we should highlight the fact that the both the data planes will be active for a short period of time during the switchover.

---
# slug: "/docs/user-guides/how-to-guides/switch-to-rudderstack-hosted-data-plane"
title: "How to Switch from Self-hosted to RudderStack-hosted Data Plane"
description: >-
  Step-by-step guide on switching from self-hosted data plane to RudderStack-hosted data plane.
---

The steps in this guide assume that you are using the RudderStack-hosted control plane and lists out the detailed steps to switch from self-hosted data plane to the RudderStack-hosted data plane.

## Self-hosted to RudderStack-hosted data plane 

Follow the steps below to switch from self-hosted to RudderStack-hosted data plane:

1. Update the `dataPlaneUrl` in all the SDKs to point to the RudderStack-hosted hosted data plane URL. Refer to the <Link to="/sources/sdks/">SDK documentation</Link> for details on how to obtain the data plane URL.

2. Verify that the self-hosted data plane is no longer getting any requests. One way is to check the load balancer metrics to verify if any requests are still received on your data plane.

> 3. Verify that all the events received in the self-hosted data plane are delivered. 

Reviewer: can we remove this line (or change wording from 'verify') since we are not actually giving any steps to verify. Rather we are just recommending to keep it running for a day to flush all events

<div class="infoBlock">
It is recommended to keep the self-hosted plane running for a day or two to ensure that all the events are delievered.
</div>

4. Shut down the self-hosted data plane dpending on the way of your deployment. For example, if you have deployed the data plane using helm, run the following command:

```BASH
helm delete <rudderstack-release-name>
```

<div class="infoBlock">
Replace the <code class="inline-code">rudderstack-release-name</code> with the release name of the chart you want to delete.
</div>

## Best practices

* In case of mobile SDKs, your customers need to update their mobile applications. You can keep the self-hosted data plane online until a sufficient number of customers have updated their mobile application or there are no more events coming in to the self-hosted data plane.

> Alternatively, you can also add an alias DNS record for your self-hosted data plane URL to point to the RudderStack-hosted data plane URL.

Reviewer: we can point them to the proxy requests from their url to rs hosted data plane url if we have that ready. Otherwise we can change the wording to something like 
```
Alternatively, you can run an API Gateway or Cloudflare workers to proxy the requests to your self hosted data plane URL to the URL given for RudderStack hosted data plane.
```
